public/app/core/utils/explore.ts

```diff
@@ -280,27 +306,19 @@ export function hasNonEmptyQuery<TQuery extends DataQuery = any>(queries: TQuery
 }
 
 export function calculateResultsFromQueryTransactions(
-  queryTransactions: QueryTransaction[],
+  result: any,
+  resultType: ResultType,
   datasource: any,
   graphInterval: number
 ) {
-  const graphResult = _.flatten(
-    queryTransactions.filter(qt => qt.resultType === 'Graph' && qt.done && qt.result).map(qt => qt.result)
-  );
-  const tableResult = mergeTablesIntoModel(
-    new TableModel(),
-    ...queryTransactions
-      .filter(qt => qt.resultType === 'Table' && qt.done && qt.result && qt.result.columns && qt.result.rows)
-      .map(qt => qt.result)
-  );
+  const graphResult = resultType === 'Graph' && result ? result : undefined;
+  const tableResult =
+    resultType === 'Table' && result && result.columns && result.rows
+      ? mergeTablesIntoModel(new TableModel(), result)
+      : mergeTablesIntoModel(new TableModel());
```

Comment:
This doesn't seem to work. Tried with using test datasource with a Random Walk and Table Static, but only the Graph are populated. However two legends are rendered for the graph which is wrong. Probably because result can include both series and tables now compared to before. 

This is something that will be easier handled in later refactorings so we just need to make sure that this works for now.

Note that for test datasource to even show a table in explore you need to change plugin.json so that `"tables": true,` and rebuild backend.

Reply:
Hopefully any issue with table vs timeseries should be solved by using SeriesData :)

Reply:
Very good catch, I've fixed it. See if you can repro it again.